kernel/tsc/src/lib.rs

```diff
@@ -2,9 +2,21 @@
 
 #[macro_use] extern crate log;
 extern crate pit_clock_basic;
+extern crate time;
 
 use core::sync::atomic::{AtomicUsize, Ordering};
 
+pub struct Tsc;
+
+impl time::ClockSource for Tsc {
+    type ClockType = time::Monotonic;
+
+    /// The current time according to the clock.
+    /// Monotonic clocks return an [`Instant`]
+    fn now() -> <Self::ClockType as time::ClockType>::Unit {
+        time::Instant::new(1)
```

Comment:
This returns a constant value of `1`. You need to actually read the current value of the timer's ticks; the functions for that are in this file.